akka-stream/src/main/scala/akka/stream/scaladsl/pattern/Debug.scala

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2014-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl.pattern
+
+import akka.actor.ActorSystem
+import akka.event.{ LogSource, BusLogging, Logging, LoggingAdapter }
+import akka.stream.ActorFlowMaterializer
+import akka.stream.impl.Stages
+import akka.stream.scaladsl.{ FlowOps, Source, FlowGraph, Flow }
+import akka.stream.stage._
+import akka.stream.scaladsl.FlowGraph.Implicits._
+import scala.annotation.implicitNotFound
+import scala.language.implicitConversions
+
+// TODO do we want this to be a "tag on pattern", or put debug() right on Source / Flow?
+object debug {
+
+  implicit def addDebugFlow[In, Out, Mat](f: Flow[In, Out, Mat])(implicit sys: ActorSystem): DebuggableFlow[In, Out, Mat] =
```

Comment:
That would work I think, yes: Would have to downcast to `ActorFlowMaterializer` and get the `LogSource` based on the `supervisor: ActorRef` contained in the materializer.